service/create/templates.go

```diff
@@ -27,6 +27,14 @@ const (
 . /etc/environment
 USERDATA_FILE={{.MachineType}}
 
+# Wait for S3 bucket to be available.
+s3_http_uri="https://s3.{{.Region}}.amazonaws.com/{{.S3URI}}/cloudconfig/$USERDATA_FILE"
+
+until [ $(curl --output /dev/null --silent --head --fail -w "%{http_code}" $s3_http_uri) -eq "403" ]; do
+   echo "waiting for $s3_http_uri to be available"
+   sleep 5
+done
```

Comment:
The check waits for a 403 from the S3 bucket to make sure its accessible. Improvements to my bash are very welcome! 😄

Reply:
LGTM! :) Not sure if the script in which it is included has an overall timeout, if not I would include one here so that we don't wait forever, perhaps with a number of retries like this one  https://github.com/giantswarm/shipyard/blob/master/image/run-checks.sh#L11